pkg/controller/servicemesh/memberroll/namespace_reconciler.go

```diff
@@ -242,7 +242,12 @@ func (r *namespaceReconciler) reconcileRoleBindings(namespace string, reqLogger
 		if !existingRoleBindings.Has(roleBindingName) {
 			reqLogger.Info("creating RoleBinding for mesh ServiceAccount", "RoleBinding", roleBindingName)
 			roleBinding := meshRoleBinding.DeepCopy()
-			roleBinding.SetNamespace(namespace)
+			roleBinding.ObjectMeta = metav1.ObjectMeta{
```

Comment:
Interesting. How did this ever work?

Reply:
It didn't :). I refactored this code a few commits ago and introduced this bug. The unit tests passed because the client doesn't perform the same checks as the API server does.